### PR TITLE
Add TracerProvider and TracerProviderFactory implementations for OTel SPI

### PR DESCRIPTION
**Pull request by HaloFour**
Adds an implementation of `TracerProvider` and `TracerProviderFactory` that can be used with Java SPI service registration and automatically loaded by `OpenTelemetry.getTracer()`.

### CODE REVIEW

**Comment by HaloFour** [comment]
I'm still trying to see if this will integrate with opentelemetry-instrumentation.  So far it seems that the Java agent automatically configures the SDK reference implementation

File: money-core/src/main/scala/com/comcast/money/core/formatters/FormatterPropagator.scala [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import java.util
+
+import com.comcast.money.api.SpanId
+import io.grpc.Context
+import io.opentelemetry.context.propagation.TextMapPropagator
+import io.opentelemetry.trace.{ DefaultSpan, TracingContextUtils }
+
+import scala.collection.JavaConverters._
+
+case class FormatterPropagator(formatter: Formatter) extends TextMapPropagator {
```
Comment (ptravers) [inline money-core/src/main/scala/com/comcast/money/core/formatters/FormatterPropagator.scala:28]:
```suggestion
final case class FormatterPropagator(formatter: Formatter) extends TextMapPropagator {
```

File: money-core/src/main/scala/com/comcast/money/core/formatters/FormatterPropagators.scala [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.comcast.money.core.formatters
+
+import io.opentelemetry.context.propagation.{ ContextPropagators, TextMapPropagator }
+
+case class FormatterPropagators(formatter: Formatter) extends ContextPropagators {
```
Comment (ptravers) [inline money-core/src/main/scala/com/comcast/money/core/formatters/FormatterPropagators.scala:21]:
```suggestion
final case class FormatterPropagators(formatter: Formatter) extends ContextPropagators {
```

File: money-core/src/main/scala/com/comcast/money/core/MoneyTracerProvider.scala [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.comcast.money.core
+
+import io.opentelemetry.trace
+import io.opentelemetry.trace.TracerProvider
+
+case class MoneyTracerProvider(tracer: Tracer) extends TracerProvider {
```
Comment (ptravers) [inline money-core/src/main/scala/com/comcast/money/core/MoneyTracerProvider.scala:22]:
```suggestion
final case class MoneyTracerProvider(tracer: Tracer) extends TracerProvider {
```

File: money-core/src/main/scala/com/comcast/money/core/MoneyTracerProvider.scala [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  override def get(instrumentationName: String): trace.Tracer = tracer
+  override def get(instrumentationName: String, instrumentationVersion: String): trace.Tracer = tracer
```
Comment (ptravers) [inline money-core/src/main/scala/com/comcast/money/core/MoneyTracerProvider.scala:24]: do we want to support getting a specific version? I believe this should error out if there`s a version mismatch

**Review by ptravers — COMMENTED** [review]
a few nits and actual q

File: money-core/src/main/scala/com/comcast/money/core/MoneyTracerProvider.scala (hunk 4 above)
Comment (HaloFour) [reply money-core/src/main/scala/com/comcast/money/core/MoneyTracerProvider.scala:24]: Let me check the docs on this interface to see exactly what is expected here.  Feels weird that if you're going through the trouble to set up SPI that you'd also make specific library+version requests to obtain a tracer.

Comment (HaloFour) [reply money-core/src/main/scala/com/comcast/money/core/MoneyTracerProvider.scala:24]: From the JavaDocs it mentions that the name/version is for the instrumentation library, I don't think Money would care about that.

Comment (HaloFour) [reply money-core/src/main/scala/com/comcast/money/core/MoneyTracerProvider.scala:24]: Looks like the SDK reference implementation returns a different `TracerSdk` instance for each requested instrumentation library.  It doesn't matter what the library names or versions are.  The library info is attached to the span and the exporter can inspect/use it, but that's all it does.

**Review by ptravers — APPROVED** [review]

**Pull request merged**